apps/backend/src/libs/modules/mailer/base-mailer.module.ts

```diff
@@ -0,0 +1,47 @@
+import { type Transporter } from "nodemailer";
+
+import { ErrorMessage } from "~/libs/enums/enums.js";
+import { MailError } from "~/libs/exceptions/exceptions.js";
+import { config } from "~/libs/modules/config/config.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+
+class BaseMailer {
+	private resetEmailUrl: string;
+	private sender = config.ENV.MAILER.ADDRESS;
+	private transporter: Transporter;
+
+	constructor(transporter: Transporter, resetEmailUrl: string) {
+		this.transporter = transporter;
+		this.resetEmailUrl = resetEmailUrl;
+	}
+
+	public sendResetPasswordEmail({
+		recipient,
+		token,
+	}: {
+		recipient: string;
+		token: string;
+	}): void {
+		this.transporter.sendMail(
```

Comment:
Create a separate method, which will receive subject, text, to

Reply:
Do we want to have a single method for sending emails? If that is the case I feel like having a url to reset email on BaseMailer is redundant.

Reply:
Doing something like mailer.sendMail(...emailContent, mailer.resetEmailIUrl) feels wrong

Reply:
current implementation is like that:
<img width="850" alt="Screenshot 2024-09-01 at 12 21 23" src="https://github.com/user-attachments/assets/b5acec58-5f99-40eb-9926-6ec2f5993092">
I know that in previous comments you said that link should be sent as an argument to the class, but if we make a generic sendEmail method, it doesn't really make sense to save the link inside of class in my opinion